src/plugins/utils/reduceObj.js

```diff
@@ -0,0 +1,17 @@
+const reduceObj = function(jsSpec, obj) {
+  if (obj['$ref']) {
+    var objPath = obj['$ref'].split('/');
+    objPath.shift();
+    return reduceObj(jsSpec, objPath.reduce((spec,p) => {return spec && spec[p]}, jsSpec));
+  } else if (obj['allOf']) {
+    var newObj = {};
+    obj['allOf'].forEach((childObj) => {
+      newObj = Object.assign(newObj, reduceObj(jsSpec, childObj));
+    });
+    return newObj;
+  }
+  return obj;
+};
+
+module.exports = reduceObj;
+  
```

Comment:
I believe there is an easier way. In other validators, we use `lodash/at` to resolve references. See [this line](https://github.com/IBM/openapi-validator/blob/d1ed5c0eef42f95aaac6befc80ae02a9fdb5bb04/src/plugins/validation/2and3/semantic-validators/walker-ibm.js#L36) for an example. You should be able to do this with the `jsSpec`. You may need some additional logic but you can just add a simple helper function in `items-required-for-array-objects.js` rather than creating a new utility file.

Reply:
That's what I was looking for, great! Although it didn't actually reduce my logic much, I still have to convert the path from `"#/path/to/ref"` to `['path', 'to', 'ref']` and then reduce any $refs or allOfs